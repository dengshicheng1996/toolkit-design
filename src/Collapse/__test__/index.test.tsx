import * as React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import {
  unstable_Collapse as Collapse,
  unstable_List as List,
} from '@ant-design/mobile'

describe('Collapse', () => {
  it('snapshot', () => {
    const component = shallow(
      <Collapse onChange={key => console.log('key', key)} defaultActiveKey="0">
        <Collapse.Panel header="Title 1">
          <List className="my-list">
            <List.Item>content 1</List.Item>
            <List.Item>content 2</List.Item>
            <List.Item>content 3</List.Item>
          </List>
        </Collapse.Panel>
        <Collapse.Panel header="Title 2">
          this is panel content2 or other
        </Collapse.Panel>
        <Collapse.Panel header="Title 3">
          text text text text text text text text text text text text text text
          text
        </Collapse.Panel>
      </Collapse>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })
})
