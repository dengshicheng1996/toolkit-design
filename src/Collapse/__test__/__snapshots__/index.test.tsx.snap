// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse snapshot 1`] = `
<ErrorBoundary
  onError={[Function]}
>
  <Collapse
    defaultActiveKey="0"
    onChange={[Function]}
  >
    <Panel
      header="Title 1"
    >
      <List
        className="my-list"
      >
        <Item
          align="middle"
          error={false}
          platform="ios"
          wrap={false}
        >
          content 1
        </Item>
        <Item
          align="middle"
          error={false}
          platform="ios"
          wrap={false}
        >
          content 2
        </Item>
        <Item
          align="middle"
          error={false}
          platform="ios"
          wrap={false}
        >
          content 3
        </Item>
      </List>
    </Panel>
    <Panel
      header="Title 2"
    >
      this is panel content2 or other
    </Panel>
    <Panel
      header="Title 3"
    >
      text text text text text text text text text text text text text text text
    </Panel>
  </Collapse>
</ErrorBoundary>
`;
